stdlib/public/core/StringBridge.swift

```diff
@@ -177,6 +179,7 @@ extension String {
       count: length,
       elementShift: isUTF16 ? 1 : 0,
       hasCocoaBuffer: true,
+      hasNulTerminator: false,
```

Comment:
It might, but I'm not sure what part of the “is-it-NUL-terminated?“ tracking code and/or actual NUL termination logic is causing the failures and apparent memory corruption.  So I am not even confident that if we set that bit, we're not breaking something, much less can reliably use it later.  There's too much 🍝  here.